### PR TITLE
refactor: remove stylevariables outside of theme

### LINKED ISSUE #467: Remove non-theme constants from styleVariables.js

### PR DESCRIPTION
**Pull request by shanberg**
- Remove all styleVariables values that aren't part of a theme
- Refactor components where necessary to pull from theme instead of styleVariables

Closes #467

### CODE REVIEW

File: src/components/AppFooter/components/__snapshots__/Copyright.test.js.snap [hunk 1]
```diff
@@ -12,8 +12,8 @@ exports[`Copyright should render when screen is 800px or higher 1`] = `
   transition: opacity 0.2s ease;
   cursor: default;
   border-top: 1px solid;
-  margin-top: 8px;
-  padding: 8px;
+  margin-top: NaNpx;
+  padding: NaNpx;
```
Comment (kaitmore) [inline src/components/AppFooter/components/__snapshots__/Copyright.test.js.snap]: `NaNpx` 👀

File: src/components/AppFooter/components/__snapshots__/Links.test.js.snap [hunk 2]
```diff
@@ -6,7 +6,7 @@ exports[`Link should render 1`] = `
   -ms-flex: 1 0 auto;
   flex: 1 0 auto;
   text-align: right;
-  padding: 0 0 4px 8px;
+  padding: 0 0 NaNpx NaNpx;
```
Comment (kaitmore) [inline src/components/AppFooter/components/__snapshots__/Links.test.js.snap]: 👀

File: src/components/Button/components/ButtonWrap/utils/generateButtonSize.test.js [hunk 3]
```diff
@@ -2,32 +2,10 @@ import generateButtonSize from "./generateButtonSize";
...
-    expect(generateButtonSize("normal")).toContain(`
-    font-size: 14px;
-    border-radius: 4px;
-    padding: 2px 8px;
-  `);
+    expect(generateButtonSize("xs")).toEqual(expect.anything());
+    expect(generateButtonSize("sm")).toEqual(expect.anything());
+    expect(generateButtonSize("lg")).toEqual(expect.anything());
+    expect(generateButtonSize("xl")).toEqual(expect.anything());
+    expect(generateButtonSize("normal")).toEqual(expect.anything());
```
Comment (kaitmore) [inline src/components/Button/components/ButtonWrap/utils/generateButtonSize.test.js]: haha expect.anything() 😆i feel like this should be on a t-shirt or motivational poster

File: src/components/ButtonGroup/__snapshots__/ButtonGroup.test.js.snap [hunk 4]
```diff
@@ -12,7 +12,7 @@ exports[`ButtonGroup matches snapshot 1`] = `
 }
 
 .c0 * + * {
-  margin-left: 4px;
+  margin-left: NaNpx;
```
Comment (kaitmore) [inline src/components/ButtonGroup/__snapshots__/ButtonGroup.test.js.snap]: 👁

File: src/components/Form/scenes/Checkbox/components/__snapshots__/Wrapper.test.js.snap [hunk 5]
```diff
@@ -22,7 +22,7 @@ exports[`Wrapper should render 1`] = `
 }
 
 .c0 > input {
-  margin-left: 8px;
+  margin-left: props => Math.round(parseInt(props.theme.SPACING_BASE,10) * factor),px;
```
Comment (kaitmore) [inline src/components/Form/scenes/Checkbox/components/__snapshots__/Wrapper.test.js.snap]: these snapshots make me sad

File: src/style/styleFunctions.js [hunk 6]
```diff
@@ -1,44 +1,7 @@
...
-  } else {
-    let enhancedContrast = contrast * DARK_ON_LIGHT_CONTRAST_ENHANCEMENT_RATIO; // DARK_ON_LIGHT_CONTRAST_ENHANCEMENT_RATIO = 2
-    intentColor = intentColor || "#FFF"; //white
-    if (enhancedContrast > 1) {
-      enhancedContrast = 1;
-    }
-    return mix(enhancedContrast, intentColor, backgroundColor);
-  }
-}
+//
```
Comment (kaitmore) [inline src/style/styleFunctions.js]: zombie

File: src/style/styleVariables.js [hunk 7]
```diff
@@ -1,68 +1,4 @@
...
-export const FONT_SIZE_XS = "9px"; // Will be changed to 10px
-
-export const LINE_HEIGHT_BASE = 1.4;
-export const FONT_WEIGHT_BASE = 400;
-export const FONT_WEIGHT_SEMIBOLD = FONT_WEIGHT_BASE * 1.25; // Will be changed to 600
-export const FONT_WEIGHT_BOLD = 600; // Will be changed to 700 or 800
-
-// DEPRECATE
-export const DARK_ON_LIGHT_CONTRAST_ENHANCEMENT_RATIO = 2;
+// export const SPACING_BASE = 8;
```
Comment (kaitmore) [inline src/style/styleVariables.js]: should this be commented out?

File: src/components/Button/components/ButtonWrap/utils/generateButtonSize.test.js (hunk 3 above)
Comment (shanberg) [reply src/components/Button/components/ButtonWrap/utils/generateButtonSize.test.js]: I'm thinking the test here should be.... (font?) size of `sm` is smaller than or equal to size of `normal`, is smaller than or equal to size of `lg`... etc.

The tests before `.anything()` were just snapshots, which was a little silly.

**Comment by shanberg** [comment]
@kaitmore 
`SPACING_SCALE` seems to be a tough issue here.

Some components appear to have issues pulling in the `styleFunctions` utility.

If there's a better way to do this I'm all ears.

**Pull request merged**